plugins/com.google.cloud.tools.eclipse.appengine.localserver.test/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerBehaviourTest.java

```diff
@@ -38,24 +38,64 @@
   @Mock private PortProber portProber;
   @Mock private IServer server;
 
-  @Test
-  public void testAdminPort_initializedTo8000() {
-    assertEquals(8000, serverBehavior.adminPort);
+  private void setUpServerPortAttribute(int serverPort) {
+    when(server.getAttribute(eq("appEngineDevServerPort"), anyInt())).thenReturn(serverPort);
+    when(server.getAttribute(eq("appEngineDevServerPort"), anyString()))
+        .thenReturn(String.valueOf(serverPort));
+    when(server.getAttribute(eq("appEngineDevServerAdminPort"), anyInt())).thenReturn(8000);
+    when(server.getAttribute(eq("appEngineDevServerAdminPort"), anyString())).thenReturn(null);
+  }
+
+  private void setUpPortAttributes(int serverPort, int adminPort) {
+    setUpServerPortAttribute(serverPort);
```

Comment:
Given that you're commenting the purpose of the arguments (`/*serverPort*/` vs `/*adminPort*/) perhaps you should just split these into `setupServerPortAttribute()` and `setupServerAdminPortAttribute()`?